### ISSUE TITLE
cpp type parsing

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#696]
Sphinx has trouble parsing C++ template types that have numeric template arguments. For example, declaring the following function within a C++ class gives a "WARNING: Invalid definition: expected name [error at 44]":

```
.. function::     void operator()(const boost::array<VertexID,2>& v)
```

while replacing it with

```
.. function::     void operator()(const boost::array<VertexID,D>& v)
```

gives no warning.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/696
- Originally reported by: Dmitriy Morozov
- Originally created at: 2011-05-11T20:40:14.873

**Comment by shimizukawa** [comment]
_From Armin Ronacher on 2011-09-22 08:23:55+00:00_

Added support for non-name template arguments.  This fixes #696

→ <<cset bdf58164379a27fb902bf7693d416dde1a813878>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-23 06:04:54+00:00_

Added support for non-name template arguments.  This fixes #696

→ <<cset bdf58164379a27fb902bf7693d416dde1a813878>>
